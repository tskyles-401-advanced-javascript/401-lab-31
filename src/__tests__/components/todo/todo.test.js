import React from 'react';
import Todo from '../../../components/todo/todo';

describe('<Todo />', () => {
  it('is alive on start', () => {
    let app = shallow(<Todo/>);

    expect(app.find('header').exists()).toBeTruthy();
    expect(app.find('.todo').exists()).toBeTruthy();
  });

  

  it('renders correctly', () => {
    const tree = renderer.create(<Todo/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});