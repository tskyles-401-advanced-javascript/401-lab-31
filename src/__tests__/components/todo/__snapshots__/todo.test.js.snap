// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> renders correctly 1`] = `
Array [
  <header>
    <h2>
      There are
      0
      Items To Complete
    </h2>
  </header>,
  <section
    className="todo"
  >
    <div>
      <h3>
        Add Item
      </h3>
      <form
        onSubmit={[Function]}
      >
        <label>
          <span>
            To Do Item
          </span>
          <input
            name="text"
            onChange={[Function]}
            placeholder="Add To Do List Item"
          />
        </label>
        <label>
          <span>
            Difficulty Rating
          </span>
          <input
            defaultValue="3"
            max="5"
            min="1"
            name="difficulty"
            onChange={[Function]}
            type="range"
          />
        </label>
        <label>
          <span>
            Assigned To
          </span>
          <input
            name="assignee"
            onChange={[Function]}
            placeholder="Assigned To"
            type="text"
          />
        </label>
        <label>
          <span>
            Due
          </span>
          <input
            name="due"
            onChange={[Function]}
            type="date"
          />
        </label>
        <button>
          Add Item
        </button>
      </form>
    </div>
    <div>
      <ul />
    </div>
  </section>,
]
`;
